### PR TITLE
Mainnet-fork compatibility

### PR DESCRIPTION
**Pull request by Padraic-O-Mhuiris**
Related to https://github.com/element-fi/elf-frontend-testnet/pull/1

The gist of this is to utilise the mainnet addresses and tokenlist whilst targeting a localhost environment. I've added instructions in the readme which should explain what is necessary to set it all up

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,17 +32,56 @@ npm start
...
+```bash
+npm run start:dev-mainnet-fork-app
+```
+
+This will spin up the frontend as normal. It will be noticeably slower in initial instances as hardhat will have to first retrieve real state but will be cached in future instances. This will repeat itself when the target blocknumber is changed.
+
+##### Metamask related issues
+
+1. Sometimes metamask can be out of sync and will attempt to cache on the basis of blocknumber last seen. This is known as ["tagging"](https://ethereum.stackexchange.com/questions/109625/received-invalid-block-tag-87-latest-block-number-is-0) and will most likely occur when the node is restarted, resetting the blocknumber. This can be circumvented as instructed from the link and so it is advised to set your network to some other and back again to localhost
+2. When submitting transactions, there may be a nonce issue which is also due to metamask caching. This can be annoying but can be manually edited with the correct one. Usually if a transaction is attempted there should be an error log in the console which should mention the correct one.
```
Comment (DannyDelott) [inline README.md:76]: In my experience doing a Hard Reset on metamask also solves the busted nonce issue: 
image

File: src/efi/addresses/addresses.ts [hunk 2]
```diff
@@ -12,20 +12,33 @@ function getAddressesJsonId() {
...
   return process.env.NEXT_PUBLIC_CHAIN_NAME || "testnet";
 }
 
 function getAddressesJson(): AddressesJsonFile {
   if (addressesJsonId === "testnet") {
-    return AddressesJsonFileTestnet;
+    return {
+      ...AddressesJsonFileMainnet,
+      chainId: AddressesJsonFileTestnet.chainId,
+    }; //AddressesJsonFileTestnet;
```
Comment (DannyDelott) [inline src/efi/addresses/addresses.ts:27]: Is the intent to move off of the locally deployed contracts in favor of a mainnet fork? I wonder if it's worth keeping these separate, ie: `testnet.addresses.json` and `mainnet_fork.addresses.json`

**Review by DannyDelott — APPROVED** [review]
Nice, much appreciated for the updated README

File: src/efi/addresses/addresses.ts (hunk 2 above)
Comment (Padraic-O-Mhuiris) [reply src/efi/addresses/addresses.ts:27]: Well I'm only using it to manipulate the chainId, it is entirely meant to sync with the mainnet environment so making another set of addresses would just be extra work to keep track of

File: README.md (hunk 1 above)
Comment (Padraic-O-Mhuiris) [reply README.md:76]: Will mention

**Pull request merged**